tests/ignite/contrib/metrics/test_roc_auc.py

```diff
@@ -1,117 +1,139 @@
 
+def test_input_types():
+    roc_auc = ROC_AUC()
+    roc_auc.reset()
+    output1 = (torch.rand(4, 3), torch.randint(0, 2, size=(4, 3), dtype=torch.long))
+    roc_auc.update(output1)
 
-def test_roc_auc_score_2():
+    with pytest.raises(ValueError, match=r"Incoherent types between input y_pred and stored predictions"):
+        roc_auc.update((torch.randint(0, 5, size=(4, 3)), torch.randint(0, 2, size=(4, 3))))
 
-    np.random.seed(1)
-    size = 100
-    np_y_pred = np.random.rand(size, 1)
-    np_y = np.zeros((size,), dtype=np.long)
-    np_y[size // 2 :] = 1
-    np.random.shuffle(np_y)
-    np_roc_auc = roc_auc_score(np_y, np_y_pred)
+    with pytest.raises(ValueError, match=r"Incoherent types between input y and stored targets"):
+        roc_auc.update((torch.rand(4, 3), torch.randint(0, 2, size=(4, 3)).to(torch.int32)))
 
-    roc_auc_metric = ROC_AUC()
-    y_pred = torch.from_numpy(np_y_pred)
-    y = torch.from_numpy(np_y)
+    with pytest.raises(ValueError, match=r"Incoherent types between input y_pred and stored predictions"):
+        roc_auc.update((torch.randint(0, 2, size=(10,)).long(), torch.randint(0, 2, size=(10, 5)).long()))
 
-    roc_auc_metric.reset()
-    n_iters = 10
-    batch_size = size // n_iters
-    for i in range(n_iters):
-        idx = i * batch_size
-        roc_auc_metric.update((y_pred[idx : idx + batch_size], y[idx : idx + batch_size]))
 
-    roc_auc = roc_auc_metric.compute()
+def test_check_shape():
+    roc_auc = ROC_AUC()
 
-    assert roc_auc == np_roc_auc
+    with pytest.raises(ValueError, match=r"Predictions should be of shape"):
+        roc_auc._check_shape((torch.tensor(0), torch.tensor(0)))
 
+    with pytest.raises(ValueError, match=r"Predictions should be of shape"):
+        roc_auc._check_shape((torch.rand(4, 3, 1), torch.rand(4, 3)))
 
-def test_integration_roc_auc_score_with_output_transform():
+    with pytest.raises(ValueError, match=r"Targets should be of shape"):
+        roc_auc._check_shape((torch.rand(4, 3), torch.rand(4, 3, 1)))
 
-    np.random.seed(1)
-    size = 100
-    np_y_pred = np.random.rand(size, 1)
-    np_y = np.zeros((size,), dtype=np.long)
-    np_y[size // 2 :] = 1
-    np.random.shuffle(np_y)
 
-    np_roc_auc = roc_auc_score(np_y, np_y_pred)
+def test_binary_input_N():
 
-    batch_size = 10
+    roc_auc = ROC_AUC()
 
-    def update_fn(engine, batch):
-        idx = (engine.state.iteration - 1) * batch_size
-        y_true_batch = np_y[idx : idx + batch_size]
-        y_pred_batch = np_y_pred[idx : idx + batch_size]
-        return idx, torch.from_numpy(y_pred_batch), torch.from_numpy(y_true_batch)
+    def _test(y_pred, y, n_iters):
+        roc_auc.reset()
+        roc_auc.update((y_pred, y))
 
-    engine = Engine(update_fn)
+        np_y = y.numpy()
+        np_y_pred = y_pred.numpy()
 
-    roc_auc_metric = ROC_AUC(output_transform=lambda x: (x[1], x[2]))
-    roc_auc_metric.attach(engine, "roc_auc")
+        if n_iters > 1:
```

Comment:
I think we already discussed about this approach and changing it by `batch_size > 1`

Reply:
But this is in `test_binary_input_n()`, it makes sense! You will find it edited properly in `test_multilabel_input_n()`

Reply:
> But this is in test_binary_input_n(), it makes sense!

I'm not sure to understand why it makes sense here.

Reply:
We can do it the both ways no problem! And i can edit it if needed..